### PR TITLE
Author ization

### PR DESCRIPTION
**Pull request by jwosborn**
Implements Users and Authorizes them to use app. Currently All users see all tasks. Features to come. This update simply requires users to create an account and log in before using the app.

### CODE REVIEW

File: app/Http/Controllers/LogoutController.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+
+class LogoutController extends Controller
+{
+    public function logout(Request $request) {
+        $session =  session('username');
```
Comment (jwhittle933) [inline app/Http/Controllers/LogoutController.php]: Is this needed?

File: app/Http/Controllers/LoginController1.php [hunk 2]
```diff
@@ -12,28 +13,26 @@ public function index()
...
-        return redirect('/login')->with('error', $error);
-    } elseif (Hash::check($loginPassword, user[0]->password)){
-        $user->isLoggedIn=1;
-        return redirect('/');
+        if(!$user){
+            $error='Invalid Username';
+            Session()->put('error', $error);
+            return redirect('/login');
+        } elseif (Hash::check($loginPassword, $user['password'])){
+            Session()->put('username', $user['username']);
```
Comment (jwhittle933) [inline app/Http/Controllers/LoginController1.php]: `$user` is an Object, not an array. You should use `$user->password`

File: app/Http/Controllers/NewUserController.php [hunk 3]
```diff
@@ -15,12 +15,13 @@ public function index()
...
-        $user->username=$request->email;
-        $user->email=$request->email;
-        $user->password=Hash::make($request['password']);
+        $user->username=$request->input('username');
+        $user->email=$request->input('email');
+        $hashed=Hash::make($request->input('password'));
+        $user->password=$hashed;
         $user->isLoggedIn=1;
         $user->save();
+        Session()->put('username', $user['username']);
```
Comment (jwhittle933) [inline app/Http/Controllers/NewUserController.php]: See above

File: config/session.php [hunk 4]
```diff
@@ -33,7 +33,7 @@
 
     'lifetime' => env('SESSION_LIFETIME', 120),
 
-    'expire_on_close' => false,
+    'expire_on_close' => true,
```
Comment (jwhittle933) [inline config/session.php]: Maybe this was an experiment for testing, but you probably don't want this

File: database/seeds/UsersTableSeeder.php [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    /**
+     * Run the database seeds.
+     *
+     * @return void
+     */
+    public function run()
+    {
+        DB::table('users')->insert([
+            'userName' => Str::random(10),
+            'email' => Str::random(10).'@gmail.com',
```
Comment (jwhittle933) [inline database/seeds/UsersTableSeeder.php]: Why `Str::random`? Don't you want to know the username and email for logging in?

File: routes/web.php [hunk 6]
```diff
@@ -12,9 +12,15 @@
 */
 
 Route::get('/', function () {
+    $user = Session::get('username');
+    if(!$user)
+    {
+        return redirect('/login');
+    }
```
Comment (jwhittle933) [inline routes/web.php]: I was with you when you did this, but there's a better way:
```php
if (! $user) return redirect('/login');
```

File: app/Http/Controllers/LogoutController.php (hunk 1 above)
Comment (jwosborn) [reply app/Http/Controllers/LogoutController.php]: Lol you wrote it. It's your function. But no. It isn't

File: database/seeds/UsersTableSeeder.php (hunk 5 above)
Comment (jwosborn) [reply database/seeds/UsersTableSeeder.php]: Good point. Just had no idea what I'm doing with seeders so I followed a model I found in the docs.

**Pull request merged**